lib/core/src/msParam.cpp

```diff
@@ -1022,6 +1022,34 @@ parseMspForPosInt( msParam_t* inpParam ) {
     return myInt;
 }
 
+int msp_to_string(msParam_t* inp_param, char** out_param)
+{
+    if (out_param == nullptr) {
+        log_msi::error("{}: out_param is null", __func__);
+        return SYS_NULL_INPUT;
+    }
+
+    if (inp_param == nullptr || inp_param->inOutStruct == nullptr) {
+        log_msi::error("{}: inp_param is null or _inpParam's inOutStruct is null", __func__);
+        *out_param = nullptr;
```

Comment:
Why set the out param if the caller gives you bad inputs?

_Same question for the other error cases._

Reply:
I think it was to somewhat align with the "old" behavior where the output is null under any error case. That being said, you're right in that it's probably better just to not touch it so that the return value is the one way to tell whether or not it errors...I'll change that!